subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/worker/TestEventSerializer.java

```diff
@@ -183,12 +182,19 @@ public DefaultTestFailure read(Decoder decoder) throws Exception {
             for (int i = 0; i < numOfCauses; i++) {
                 causes.add(read(decoder));
             }
+
+            String rawFailureName = decoder.readString();
+            Throwable rawFailure;
+            try {
+               rawFailure = throwableSerializer.read(decoder);
+            } catch(Exception e) {
```

Comment:
💭 Should this perhaps catch any `Throwable`?  IDK what other type of serialization failures we might encounter here.

Reply:
I think in general you should avoid catching `Throwable`s, since they include `Error`s like OOMs

Reply:
Yeah, that's definitely the downside.  But OTOH, an `OOM` _during de-serialization_ might be a result of some wacky custom de-serialization logic bug.

Reply:
Yes, potentially, but we would want that OOM to bubble up and fail the build. If we caught it here, we would be saving it off as a reference and letting execution continue, which is not what we want to do with an OOM.

Reply:
I think that's maybe debatable.  A memory issue caused by a bad de-serialization of a custom exception type might be fully recoverable rather than fatal.  But I don't feel too strongly about this, I just just noting it something to consider.

Reply:
How could we feasibly recover from this? Recovering from OOMs is not something I really have experience with, so I'm not sure what we'd do here in this case

Reply:
If the failure is caused during attempted allocation of a gigantic array during instantiation, for instance, the JVM is fine afterwards - you just don't have an instance (or a gigantic array).

Reply:
I agree with @jvandort.  I'd prefer to just catch `Exception` here for now.  We can always make this less specific if we find a case where we get a `Throwable` instead.